### ISSUE TITLE
[Faceted Filters] Missing heading structure

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#111]
## Issue summary

The Faceted Filters section is missing heading structure. Headings are crucial to conveying meaning for each section and to allow for heading navigation using assistive technology.

For example, screen reader users navigate via headings alone to gain an understanding of the current page content. Lack of heading structure will result in a frustrating user experience.

<details>
  <summary>Screenshots</summary>
<img width="832" alt="Screen shot of faceted filters section with Filter by, Sort by, and Showing text highlighted." src="https://user-images.githubusercontent.com/1392632/124606741-2b8c8c00-de3b-11eb-9c72-515637a6cab8.png">

</details>

#### HTML

```html
<p class="facets__heading">Filter by</p>

<!-- … -->

<label class="collection-filters__label" for="SortBy">Sort by</label>

<!-- … -->

<p class="collection-product-count light">Showing 74 of 74 products</p>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported browser.
2. With an active screen reader, navigate via headings alone.

### Expected

- Facet filter sections to include headings.

### Actual

- Heading structure is missing.

## Recommendation

1. Swap `p.facets__heading` for an `h2` element.
2. Wrap `label.collection-filters__label` element with an `h2`.
3. Wrap `p.collection-product-count` with an `h2` element (also swap the block-level `p` for an inline `span`.)

Now, screen reader users will be able to quickly understand the sections on the page and move between them via single keystroke.

#### HTML

```html
<h2 class="facets__heading">Filter by</h2>

<!-- … -->

<h2>
  <label class="collection-filters__label" for="SortBy">Sort by</label>
</h2>

<!-- … -->

<h2>
  <span class="collection-product-count" role="status">
    Showing 74 of 74 products
  </span>
</h2>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.10: Section Headings](https://www.w3.org/WAI/WCAG21/Understanding/section-headings.html)
- **Severity:** High
- **Effort:** Low
- **Related:** (#115, #116)
